Sources/Sentry/SentryTracer.m

```diff
@@ -101,9 +104,117 @@ @implementation SentryTracer {
 #endif // SENTRY_HAS_UIKIT
 }
 
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+static NSObject *appLaunchTraceLock;
+static SentryId *_Nullable appLaunchTraceId;
+BOOL isTracingAppLaunch;
+static uint64_t appLaunchSystemTime;
+SentryTracesSamplerDecision *appLaunchTraceSamplerDecision;
+
+/**
+ * Decision tree:
+ * ```
+ * ─ is there a configuration file?
+ *   ├── yes: SentryTracesSampler sampling decision?
+ *   │   ├── yes: SentryProfilesSampler sampling decision?
+ *   │   │   ├── yes: ✅ trace app launch
+ *   │   │   └── no: ❌ do not trace app launch
+ *   │   └── no: ❌ do not trace app launch
+ *   └── no: ❌ do not trace app launch
+ * ```
+ */
++ (BOOL)shouldTraceAppLaunch
+{
+    [SentryFileManager load];
+
+    NSString *configFilePath = [SentryFileManager sentryLaunchConfigPath];
+
+    if (![[NSFileManager defaultManager] fileExistsAtPath:configFilePath]) {
+        SENTRY_LOG_DEBUG(@"No launch profile config file.");
+        return NO;
+    }
+
+    // get app launch configuration options
+    NSError *error;
+    NSData *configData = [NSData dataWithContentsOfFile:configFilePath options:0 error:&error];
+    if (error != nil) {
+        // we can't read the config file? bail out
+        SENTRY_LOG_DEBUG(@"Couldn't read launch profile config file: %@.", error);
+        return NO;
+    }
+
+    if (configData == nil) {
+        SENTRY_LOG_DEBUG(@"The profile config file was empty.");
+        return NO;
+    }
+
+    error = nil;
+    // a map of strings to numeric values for sample rates or boolean values for
+    // enabling/disabling launch tracing
+    NSDictionary<NSString *, NSNumber *> *launchConfig =
+        [NSJSONSerialization JSONObjectWithData:configData options:0 error:&error];
```

Comment:
`m`: We could add a method for reading the appLaunchConfig data that returns the `NSDictionary` containing all this code to `SentryFileManager` as this code here has not much to do with the tracer and fits better to the `SentryFileManager`.

Reply:
I'm actually going to go the route of just writing an empty marker file, so if that's present, that means the sampling decisions led to YES for both traces/profiles, and we remove the marker file otherwise (if one exists).